### PR TITLE
DBZ-2982 Adding docs for properties that customize log mining behavior

### PR DESCRIPTION
**Pull request by mpermar**
https://issues.jboss.org/browse/DBZ-2982

Added several new properties that existed in code and that now allow customizing log mining behavior.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Hey @Naros, could you check out this one?

@mpermar, thanks! Could you please re-author the commit using an e-mail address linked to your GH account? Thx!

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 1]
```diff
@@ -1422,6 +1422,35 @@ This field is required to enable Oracle RAC support.
...
+
+|[[oracle-property-log-mining-batch-size-max]]<<oracle-property-log-mining-batch-size-max, `log.mining.batch.size.max`>>
+|100000
+|The maximum SCN interval size that this connector will use when reading from redo/archive logs.
+
+|[[oracle-property-log-mining-batch-size-default]]<<oracle-property-log-mining-batch-size-default, `log.mining.batch.size.default`>>
+|20000
+|The starting SCN interval size that the connector will use for reading data from redo/archive logs.
+
+|[[oracle-property-log-mining-sleep-time-min]]<<oracle-property-log-mining-sleep-time-min, `log.mining.sleep.time.min`>>
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1437]: The option name should end with ".ms", as it's the convention for similar duration-kind properties.

**Review by gunnarmorling — COMMENTED** [review]
One comment inline, note this will also require a code change.

**Comment by mpermar** [comment]
@gunnarmorling thanks for the heads up. Renamed the properties. Also I actually had forgot one, so added it too.

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 2]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
 Based on the used hash function referential integrity is kept while data is pseudonymized. Supported hash functions are described in the {link-java7-standard-names}[MessageDigest section] of the Java Cryptography Architecture Standard Algorithm Name Documentation.
 The hash is automatically shortened to the length of the column.
 
+|[[oracle-property-log-mining-batch-size-min]]<<oracle-property-log-mining-batch-size-min, `log.mining.batch.size.min`>>
+|1000
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1426]:
```suggestion
|`1000`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 3]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The minimum SCN interval size that this connector will try to read from redo/archive logs. Active batch size will be also increased/decreased by this amount for tuning connector throughput when needed.
+
+|[[oracle-property-log-mining-batch-size-max]]<<oracle-property-log-mining-batch-size-max, `log.mining.batch.size.max`>>
+|100000
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1430]:
```suggestion
|`100000`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 4]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The maximum SCN interval size that this connector will use when reading from redo/archive logs.
+
+|[[oracle-property-log-mining-batch-size-default]]<<oracle-property-log-mining-batch-size-default, `log.mining.batch.size.default`>>
+|20000
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1434]:
```suggestion
|`20000`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 5]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The starting SCN interval size that the connector will use for reading data from redo/archive logs.
+
+|[[oracle-property-log-mining-sleep-time-min-ms]]<<oracle-property-log-mining-sleep-time-min-ms, `log.mining.sleep.time.min.ms`>>
+|0
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1438]:
```suggestion
|`0`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 6]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The minimum amount of time that the connector will sleep after reading data from redo/archive logs and before starting reading data again. Value is in milliseconds.
+
+|[[oracle-property-log-mining-sleep-time-max-ms]]<<oracle-property-log-mining-sleep-time-max-ms, `log.mining.sleep.time.max.ms`>>
+|3000
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1442]:
```suggestion
|`3000`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 7]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The maximum amount of time that the connector will sleep after reading data from redo/archive logs and before starting reading data again. Value is in milliseconds.
+
+|[[oracle-property-log-mining-sleep-time-default-ms]]<<oracle-property-log-mining-sleep-time-default-ms, `log.mining.sleep.time.default.ms`>>
+|1000
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1446]:
```suggestion
|`1000`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 8]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The starting amount of time that the connector will sleep after reading data from redo/archive logs and before starting reading data again. Value is in milliseconds.
+
+|[[oracle-property-log-mining-sleep-time-increment-ms]]<<oracle-property-log-mining-sleep-time-increment-ms, `log.mining.sleep.time.increment.ms`>>
+|200
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1450]:
```suggestion
|`200`
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 9]
```diff
@@ -1422,6 +1422,39 @@ This field is required to enable Oracle RAC support.
...
+|The maximum amount of time up or down that the connector will use to tune the optimal sleep time when reading data from logminer. Value is in milliseconds.
+
+|[[oracle-property-log-mining-view-fetch-size]]<<oracle-property-log-mining-view-fetch-size, `log.mining.view.fetch.size`>>
+|10000
```
Comment (Naros) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1454]:
```suggestion
|`10000`
```

**Review by Naros — COMMENTED** [review]
These are some small suggestions to bring the default values for these new options to be rendered like the other configuration properties.

**Comment by mpermar** [comment]
Thanks @Naros . Added your changes.

**Comment by Naros** [comment]
Thanks, this is good to be merged once we merge the source code changes from https://github.com/debezium/debezium-incubator/pull/246

**Comment by Naros** [comment]
Thanks @mpermar!

**Comment by gunnarmorling** [comment]
Thanks all!

**Pull request merged**